### PR TITLE
EDSC-2352: Modifies token management

### PR DESCRIPTION
**Pull request by abbottry**

### CODE REVIEW

File: serverless/src/util/urs/getUserAccessToken.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  const existingUserTokens = await dbConnection('user_tokens')
+    .first([
+      'access_token',
+      'refresh_token',
+      'expires_at'
+    ])
+    .where({ user_id: id })
+    .orderBy('created_at', 'DESC')
+
+  console.log('existingUserTokens', existingUserTokens)
```
Comment (abbottry) [inline serverless/src/util/urs/getUserAccessToken.js]: Should remove this

File: serverless/src/edlAuthorizer/handler.js [hunk 2]
```diff
@@ -48,36 +52,92 @@ const edlAuthorizer = async (event) => {
...
+        console.log(`JWT Token Invalid. ${verifyError}`)
+
         throw new Error('Unauthorized')
       }
 
+      const {
+        id: userId,
+        username
+      } = decodedJwtToken
+      console.log('decodedJwtToken', decodedJwtToken)
```
Comment (abbottry) [inline serverless/src/edlAuthorizer/handler.js]: Remove this

File: serverless/src/util/urs/getAccessTokenFromJwtToken.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { getDbConnection } from '../database/getDbConnection'
+import { getVerifiedJwtToken } from '../getVerifiedJwtToken'
+
+// Knex database connection object
+let dbConnection = null
```
Comment (abbottry) [inline serverless/src/util/urs/getAccessTokenFromJwtToken.js]: I'm not convinced this provides us any value, does this value get kept in memory with this scope or are we just overwriting a potentially 'cached' value each time we call this method?

File: serverless/src/util/urs/getUserAccessToken.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+import { getDbConnection } from '../database/getDbConnection'
+
+// Knex database connection object
+let dbConnection = null
```
Comment (abbottry) [inline serverless/src/util/urs/getUserAccessToken.js]: I'm not convinced this provides us any value, does this value get kept in memory with this scope or are we just overwriting a potentially 'cached' value each time we call this method?

File: serverless/src/util/urs/getAccessTokenFromJwtToken.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Returns the decrypted urs system credentials from Secrets Manager
+ */
+export const getAccessTokenFromJwtToken = async (jwtToken) => {
+  // Retrieve a connection to the database
+  dbConnection = await getDbConnection(dbConnection)
+
+  const { id } = getVerifiedJwtToken(jwtToken)
+
+  const existingUserTokens = await dbConnection('user_tokens')
```
Comment (abbottry) [inline serverless/src/util/urs/getAccessTokenFromJwtToken.js]: Consider returning `user_id` for use in methods that require both a user id and token from the jwtToken.

File: static/src/js/util/request/logoutRequest.js [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+export default class LogoutRequest extends Request {
+  constructor(authToken) {
+    super(getEnvironmentConfig().apiHost)
+
+    this.authenticated = true
+    this.authToken = authToken
+  }
+
+  logout() {
+    return super.delete('/logout')
```
Comment (abbottry) [inline static/src/js/util/request/logoutRequest.js]: We've used `this.` for all of our other request classes, is there a difference in this call?

Comment (macrouch) [reply static/src/js/util/request/logoutRequest.js]: the one I copied from uses super 🤷‍♂ probably change this one and conceptRequest.js

**Pull request merged**